### PR TITLE
Update Upgrader interface comments to include expected system state.

### PR DESCRIPTION
**Pull request by EdColeman**
Capture discussion on PR #2185 to update javadoc comments.
Doc update only - contains no code changes.

### CODE REVIEW

**Review by ctubbsii — APPROVED** [review]

File: server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java [hunk 1]
```diff
@@ -30,9 +30,30 @@
  * complete and then be run again later.
  */
 public interface Upgrader {
+
+  /**
+   * Update entries in ZooKeeper - normally runs before the root tablet is loaded.
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java:35]:
```suggestion
   * Update entries in ZooKeeper - called before the root tablet is loaded.
```

File: server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java [hunk 2]
```diff
@@ -30,9 +30,30 @@
...
+   *
+   * @param ctx
+   *          the server context.
+   */
   void upgradeZookeeper(ServerContext ctx);
 
+  /**
+   * Update the root tablet - normally runs after the root tablet is loaded and before the metadata
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java:43]:
```suggestion
   * Update the root tablet - called after the root tablet is loaded and before the metadata
```

File: server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java [hunk 3]
```diff
@@ -30,9 +30,30 @@
...
+   * table is loaded.
+   *
+   * @param ctx
+   *          the server context.
+   */
   void upgradeRoot(ServerContext ctx);
 
+  /**
+   * Update the metadata table - normally runs after the metadata table is loaded and before loading
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader.java:52]:
```suggestion
   * Update the metadata table - called after the metadata table is loaded and before loading
```

**Review by keith-turner — APPROVED** [review]

**Comment by milleruntime** [comment]
@EdColeman you read my mind with these comments and beat me to the punch. :1st_place_medal:  Thanks for getting these in.

**Pull request merged**